compiler/luci-codegen/README.md

```diff
@@ -0,0 +1,203 @@
+# luci-codegen
+
+## Overview
+
+`luci-codegen` component is responcible for native code generation from graph representation.
+
+Package provides:
+- Library `luci_codegen`: takes luci IR, replaces some nodes with special "Compiled" custom nodes.
+- Executable `circle_codegen`: takes neural network in circle format, generates new circle model with compiled subgraphs. 
+
+## Build
+
+This component needs external `Halide` build. If `Halide` is built from sources it demands `LLVM`.
+
+To get needed components and build `luci_codegen` library and `circle_codegen` executable on linux you can use following instruction.
+Also see README.md in Halide repo for more methods to build LLVM and Halide (including windows build).
+
+##### LLVM Build
+
+Tested on commit **24faa87075ac1b4e4d7f7336271b1ba72ffcc2c0** from Jan 12 (master branch)
+
+```
+$ git clone https://github.com/llvm/llvm-project.git
+$ cd llvm-project
+# mkdir build; cd build
+$ cmake -DCMAKE_BUILD_TYPE=Release \
+        -DCMAKE_INSTALL_PREFIX=../install \
+        -DLLVM_ENABLE_PROJECTS="clang;lld;clang-tools-extra" \
+        -DLLVM_TARGETS_TO_BUILD="X86;ARM;NVPTX;AArch64;Mips;Hexagon" \
+        -DLLVM_ENABLE_TERMINFO=OFF -DLLVM_ENABLE_ASSERTIONS=ON \
+        -DLLVM_ENABLE_EH=ON -DLLVM_ENABLE_RTTI=ON -DLLVM_BUILD_32_BITS=OFF \
+        ../llvm
+$ make -j$(nproc)
+$ make install
+$ export LLVM_DIR=${PWD}../install/lib/cmake/llvm
+
+```
+
+##### Halide build
+
+Tested on commit **890a5199ae8ade759de076217670ae980bf44823** Dec 23.
+
+If you did not export variable from previous LLVM build you can pass it via cmake argument: `-DLLVM_DIR=<path to llvm cmake config>` 
+
+```
+$ git clone https://github.com/halide/Halide.git
+$ cd Halide
+$ mkdir build; cd build
+$ cmake -DCMAKE_BUILD_TYPE=Release \
+        -DCMAKE_INSTALL_PREFIX=../install \
+        ../
+$ make -j$(nproc)
+$ make install
+$ export Halide_DIR=${PWD}/../install/lib/cmake/Halide
+```
+
+##### luci-codegen build
+
+Assuming current dir is where build should be done.
+If `LLVM_DIR` and `Halide_DIR` variables are not set you can pass them as cmake arguments.
+
+```
+$ cmake <path to nncc CMakeLists.txt dir>
+$ make -j$(nproc) circle_codegen
+```
+
+`circle_codegen` executable path is `<build dir>/compiler/luci-codegen/src/circle_codegen`
+
+`luci_codegen` library path is `<build dir>/compiler/luci-codegen/src/luci_codegen.so`
+
+## Usage
+
+##### circle_codegen
+
+`luci_codegen` uses Halide autoschedulers, they are built along with main Halide library (libHalide.so on linux)
```

Comment:
Typo? 

```suggestion
`circle_codegen` uses Halide autoschedulers, they are built along with main Halide library (libHalide.so on linux)
```

Reply:
This is a misleading naming of utility (circle_codegen) and library (luci_codegen) that is part of circle_codegen.
Literally everyone is confused, sorry. I'll try to come up with more distinguishable names.